### PR TITLE
chore: Refactor `cache` into `core`

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

This change refactors `@aws-amplify/cache` into `@aws-amplify/core` and deletes the `@aws-amplify/cache` package.

Coming soon:
- Setting up initial export paths from `@aws-amplify/core` & `aws-amplify`

Other changes required to facilitate this move:
- Updated `@aws-amplify/core` package.json with required entry-points, bundle size checks, and jest config
- Refactored internal dependencies on `@aws-amplify/cache` exports
- Remapped umbrella package exports relating to cache (no surface change)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by israx — APPROVED** [review]
LGTM

File: packages/aws-amplify/src/index.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export {
 	syncExpression,
 } from '@aws-amplify/datastore';
 export { PubSub } from '@aws-amplify/pubsub';
-export { Cache } from '@aws-amplify/cache';
+export { Cache } from '@aws-amplify/core';
```
Comment (elorzafe) [inline packages/aws-amplify/src/index.ts:24]: I would move this to the other L1

File: packages/cache/CHANGELOG.md
Comment (elorzafe) [inline packages/cache/CHANGELOG.md:1]: Would we like to move this to core at least for history?

**Review by elorzafe — APPROVED** [review]
One minor import change comment

Overall looks great

**Review by elorzafe — APPROVED** [review]
Looks great!

Thanks @jimblanc 🔥

**Review by israx — APPROVED** [review]

**Review by israx — APPROVED** [review]

**Review by elorzafe — APPROVED** [review]

**Pull request merged**